packages/ember/src/components/form/select.gts

```diff
@@ -0,0 +1,445 @@
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { get, action } from '@ember/object';
+import { service } from '@ember/service';
+import { isEqual } from '@ember/utils';
+import Component from '@glimmer/component';
+// @ts-expect-error Glimmer doesn't currently ship a type for the `cached` decorator
+// https://github.com/glimmerjs/glimmer.js/issues/408
+import { tracked, cached } from '@glimmer/tracking';
+// @ts-expect-error Ember keyboard doesn't currently ship a type for the `on-key` modifier
+// https://github.com/adopted-ember-addons/ember-keyboard/issues/464
+import onKey from 'ember-keyboard/modifiers/on-key';
+import { runTask } from 'ember-lifeline';
+
+import BoundValue from './bound-value.ts';
+import onInsert from '../../modifiers/did-insert.ts';
+
+import type { Optional } from '../../types.d.ts';
+import type IntlService from 'ember-intl/services/intl';
+
+declare type SelectOption<T> = {
+  label: string;
+  value: string | T;
+  raw: T;
+};
+
+interface SelectItemSignature<T> {
+  Args: {
+    optionIndex: number;
+    activeIndex: number;
+    currentValue: string | T;
+    option: SelectOption<T>;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLLIElement;
+}
+
+class SelectItem<T> extends Component<SelectItemSignature<T>> {
+  get classList() {
+    const classes = ['dropdown-item'];
+
+    const useIndexActive = this.args.activeIndex != -1;
+    const isCurrentIndex = isEqual(
+      this.args.optionIndex,
+      this.args.activeIndex,
+    );
```

Comment:
This can still be a normal `===`, since the indices are numbers
```suggestion
    const isCurrentIndex = this.args.optionIndex === this.args.activeIndex;
```